### PR TITLE
+FEATURE: Allow to import only a specific list of attributes from Akeneo and no…

### PR DESCRIPTION
**Pull request by sylvainraye**
…t all. Use a separated value list into a specific field inside the integration settings form.

### CODE REVIEW

File: Entity/AkeneoSettings.php [hunk 1]
```diff
@@ -608,4 +616,24 @@ public function setPriceList(PriceList $priceList): self
...
+    {
+        return $this->attributesList;
+    }
+
+    /**
+     * @param string $attributeList
+     *
+     * @return $this
+     */
+    public function setAttributesList($attributeList)
```
Comment (dxops) [inline Entity/AkeneoSettings.php]: Add typehint and return type please

File: Form/Type/AkeneoSettingsType.php [hunk 2]
```diff
@@ -411,6 +419,8 @@ public function onPreSubmit(FormEvent $event)
                 ]
             );
 
+            $transportData['attributesList'] = str_replace(' ','', $transportData['attributesList']);
```
Comment (dxops) [inline Form/Type/AkeneoSettingsType.php]: On Akeneo side, `Attribute code may contain only letters, numbers and underscores`, so makes sense to validate attribute names also

File: Integration/AkeneoTransport.php [hunk 3]
```diff
@@ -246,11 +248,18 @@ public function getLabel()
 
     /**
      * @param int $pageSize
+     * @param null $filter
```
Comment (dxops) [inline Integration/AkeneoTransport.php]: Remove this line please

File: Integration/AkeneoTransport.php [hunk 4]
```diff
@@ -246,11 +248,18 @@ public function getLabel()
...
      *
      * @return \Iterator
      */
     public function getAttributes(int $pageSize)
     {
-        return new AttributeIterator($this->client->getAttributeApi()->all($pageSize), $this->client, $this->logger);
+        $attributeFilter = [];
+        $attrList = $this->transportEntity->getAttributesList();
+        if (!empty($attrList)) {
+            $attributeFilter = explode(';', str_replace(' ', '', $attrList));
```
Comment (dxops) [inline Integration/AkeneoTransport.php]: It looks like `$this->transportEntity->getAttributesList()` loaded without spaces here

File: Integration/Iterator/AttributeIterator.php [hunk 5]
```diff
@@ -54,4 +94,29 @@ function ($a, $b) {
             }
         );
     }
+
+    /**
+     * @param $attribute
```
Comment (dxops) [inline Integration/Iterator/AttributeIterator.php]: Fix phpdoc and add return type please

**Review by dxops — CHANGES_REQUESTED** [review]
Hi, @sylvainraye ,

thank you for your contribution, check my comments below please

File: Entity/AkeneoSettings.php (hunk 1 above)
Comment (sylvainraye) [reply Entity/AkeneoSettings.php]: done

File: Form/Type/AkeneoSettingsType.php (hunk 2 above)
Comment (sylvainraye) [reply Form/Type/AkeneoSettingsType.php]: done

File: Integration/AkeneoTransport.php (hunk 3 above)
Comment (sylvainraye) [reply Integration/AkeneoTransport.php]: done

File: Integration/AkeneoTransport.php (hunk 4 above)
Comment (sylvainraye) [reply Integration/AkeneoTransport.php]: got issue previously, but I changed it. done

File: Integration/Iterator/AttributeIterator.php (hunk 5 above)
Comment (sylvainraye) [reply Integration/Iterator/AttributeIterator.php]: done

**Comment by dxops** [comment]
Hello, @sylvainraye , form updates are missing, so it's not possible to create an integration.

Can we merge this PR with #24 also, to decrease code review and QA time?

**Comment by sylvainraye** [comment]
Hi, yes you can

**Comment by dxops** [comment]
@sylvainraye what about form fields? I get fatal errors on integration form now

**Comment by sylvainraye** [comment]
What are the errors you get ?

**Comment by dxops** [comment]
https://github.com/oroinc/OroAkeneoBundle/pull/22/files#diff-460e769f42ed6a8f061c31d97916f203R188-R192

child name is wrong, but I see the valid one, so will be able to fix it, thanks

**Comment by dxops** [comment]
Fixed by https://github.com/oroinc/OroAkeneoBundle/commit/2a654776c0bbafb1fffc8059686fe9c8c9bf7ee7

Thanks @sylvainraye !

**Pull request closed without merging**